src/workerd/api/sockets.c++

```diff
@@ -255,8 +260,12 @@ void Socket::handleProxyStatus(
     if (status.statusCode < 200 || status.statusCode >= 300) {
       // If the status indicates an unsucessful connection we need to reject the `closeFulfiller`
       // with an exception. This will reject the socket's `closed` promise.
+      auto msg = kj::str(": proxy request failed, cannot connect to the specified address");
+      if (port == 443 || port == 80) {
+        msg = kj::str(msg, ", consider using fetch instead");
```

Comment:
Maybe a bit more context about why we're suggesting using `fetch()` could be helpful?

```suggestion
        msg = kj::str(msg, ". It looks like you might be trying to connect to a HTTP-based service — consider using fetch instead");
```